server/utils/serviceProviderCSVGenerator.test.ts

```diff
@@ -0,0 +1,53 @@
+import ServiceProviderCSVGenerator from './serviceProviderCSVGenerator'
+
+describe(ServiceProviderCSVGenerator, () => {
+  describe('.generate', () => {
+    describe('with valid ServiceProviderReportReferrals', () => {
+      it('returns a CSV-formatted string', () => {
+        const reportingResponse = [
+          {
+            referralLink: 'https://refer-and-monitor.com/referral/c9f9e22f-ddd9-422a-a9af-76fc02b18b38',
+            referralRef: 'SM1973AC',
+            referralId: 'c9f9e22f-ddd9-422a-a9af-76fc02b18b38',
+            contractId: 'thing',
+            organisationId: 'XYZ5678',
+            referringOfficerEmail: 'joe.probation@example.com',
+            caseworkerId: 'liane.supplier@example.com',
+            serviceUserCRN: 'X017844',
+            dateReferralReceived: '2021-06-27T14:49:01+01:00',
+            dateSAABooked: '2021-06-27T14:49:01+01:00',
+            dateSAAAttended: '2021-06-27T14:49:01+01:00',
```

Comment:
I think we should also test how the CSV looks when some of the nullable columns are `null`.